lib/nodeUtils.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function toJson(node) {
```

Comment:
Depending on how often this function is invoked and what V8 does to potentially optimize, this might not be ideal. Each time `toJson()` is called, these inner functions are recreated. Since this is now a module, and the only thing that is visible from outside is `toJson`, then it probably makes most sense to move these functions out from the body of `toJson`.